### PR TITLE
Tip of the day (Hackdays project)

### PR DESCRIPTION
**Pull request by andyw8**
with @mllemango, @thecodepixi and @tiffanytse 

### WHAT is this pull request doing?
image

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR

### CODE REVIEW

**Comment by paulomarg** [comment]
Random comment: have we considered getting the tips from a remote location? Maybe add them as a file on github and poll that for the tips?

**Comment by thecodepixi** [comment]
> Random comment: have we considered getting the tips from a remote location? Maybe add them as a file on github and poll that for the tips?

We have! One of our goals is to have a Tip of the Day API to pull from (or at the very least some remote file).

File: lib/shopify-cli/tip_of_the_day.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          http.read_timeout = 5 # seconds
+          http.use_ssl = true
+          response = http.request_get(remote_uri.path)
+        rescue *NETWORK_ERRORS
+          return
+        end
+        unless response.is_a?(Net::HTTPSuccess)
+          return
+        end
+        tips_content = response.body
```
Comment (andyw8) [inline lib/shopify-cli/tip_of_the_day.rb:84]: @mllemango @thecodepixi We should probably add some validation here to check the response is what we expect, e.g. some JSON containing a `tips` key.

**Comment by tiffanytse** [comment]
Can we use a different color for links - like the same blue as the outline? the other blue is a bit hard to read.

File: lib/shopify-cli/tip_of_the_day.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
+require 'json'
+require 'net/http'
+
+module ShopifyCli
+  class TipOfTheDay
+    WEEK_IN_SECONDS = 7 * 24 * 60 * 60
+    DAY_IN_SECONDS = 24 * 60 * 60
+    TIPS_URL = "https://raw.githubusercontent.com/Shopify/shopify-app-cli/tip-of-the-day/docs/tips.json"
```
Comment (andyw8) [inline lib/shopify-cli/tip_of_the_day.rb:8]: (after this is merged we'll update this to point to master).

File: lib/shopify-cli/tip_of_the_day.rb (hunk 1 above)
Comment (andyw8) [reply lib/shopify-cli/tip_of_the_day.rb:71]: I've added code to handle this.

**Comment by pepicrft** [comment]
Hey @andyw8, is this still relevant?
I'm doing some clean up work on the repository and noticed that this PR has been open for a while.

**Pull request closed without merging**